### PR TITLE
test(client): test relationMode with @ map and @@ map

### PR DESCRIPTION
**Pull request by Jolg42**
Related: https://github.com/prisma/prisma/issues/15733

Issue opened from [failures](https://github.com/prisma/prisma/actions/runs/3242367144/jobs/5315529883#step:11:1462):
https://github.com/prisma/prisma/issues/15776

### CODE REVIEW

File: packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/tests_m-to-n-MongoDB.ts [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+        // Note: it's not possible on MongoDB to mutate _id, it is immutable
+        test.failing('[update] id (_id) should throw at runtime because id field is read-only/immutable', async () => {
+          await expect(
+            prisma[postModel].update({
+              where: {
+                id: '1',
+              },
+              data: {
+                // This would show a type error
```
Comment (millsp) [inline packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/tests_m-to-n-MongoDB.ts:181]: We have magic comments like for conditional type checks
`// @ts-test-if: provider !== 'mongodb'`

Comment (Jolg42) [reply packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/tests_m-to-n-MongoDB.ts:181]: But we don't ignore it here, so adding this condition won't change anything?

Comment (Jolg42) [reply packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/tests_m-to-n-MongoDB.ts:181]: This code is only running for mongodb here

**Review by jkomyno — APPROVED** [review]
🚢

File: packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/prisma/_schema_m_to_n.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+//   categories CategoryManyToMany[]
+//   published   Boolean?
+// }
+
+// model CategoryManyToMany {
+//   id    String    @id
+//   posts PostManyToMany[]
+//   published   Boolean?
+// }
+// `
```
[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/prisma/_schema_m_to_n.ts:89]: Probably not

[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/client/tests/functional/relationMode-m-n-mongodb-failing-with-at-map/prisma/_schema_m_to_n.ts:89]: Removed!

**Pull request merged**